envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/AccessLogFilter.kt

```diff
@@ -73,24 +79,110 @@ class AccessLogFilter(
             .build()
     }
 
-    private fun AccessLog.Builder.buildFromSettings(settings: StatusCodeFilterSettings) {
-        this.setFilter(
-            AccessLogFilter.newBuilder().setStatusCodeFilter(
-                StatusCodeFilter.newBuilder()
-                    .setComparison(
-                        ComparisonFilter.newBuilder()
-                            .setOp(settings.comparisonOperator)
-                            .setValue(
-                                RuntimeUInt32.newBuilder()
-                                    .setDefaultValue(settings.comparisonCode)
-                                    .setRuntimeKey("access_log_filter_http_code")
-                                    .build()
-                            )
-                            .build()
-                    )
+    private fun AccessLog.Builder.buildFromSettings(settings: AccessLogFilterSettings) {
+        val accessLogFilters = mutableListOf<AccessLogFilter?>()
+        accessLogFilters += settings.statusCodeFilterSettings?.let {
+            createStatusCodeFilter(it)
+        }
+        accessLogFilters += settings.durationFilterSettings?.let {
+            createDurationFilter(it)
+        }
+        accessLogFilters += settings.notHealthCheckFilter?.let {
+            if (it) {
+                AccessLogFilter.newBuilder().setNotHealthCheckFilter(NotHealthCheckFilter.newBuilder()).build()
+            } else {
+                null
+            }
+        }
+        accessLogFilters += settings.responseFlagFilter?.let {
+            createResponseFlagFilter(it)
```

Comment:
Why you add null and in line 109 you removed them. I think better option would be to add item in `let` block and then you don't need to worry about nulls

Reply:
i changed it for all filter for consistency